### PR TITLE
PIM-2743 : keep page per view on product grid

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                    | A
| -------------------- | ---
| Bug fix?             | 
| New feature?         |y
| BC breaks?           |n
| CI currently passes? |y
| Tests pass?          |y
| Scenarios pass?      |y
| Checkstyle issues?*  |n
| PMD issues?**        |n
| Changelog updated?   |y
| Fixed tickets        | PIM-2743
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Resources/views/macros.html.twig [hunk 1]
```diff
@@ -36,12 +36,15 @@
                                     'ASC' :
                                     'DESC';
                         }
+
+                        if (undefined != filters.pageSize) {
+                            urlParams['product-grid[_pager][_per_page]'] = filters.pageSize;
```
Comment (sumbobyboys) [inline src/Pim/Bundle/DataGridBundle/Resources/views/macros.html.twig]: This file is not used only for the product-grid. Please use {{ name }} variable :)

**Comment by jmleroux** [comment]
The paginator will undergo a more deep rewrite

**Comment by jmleroux** [comment]
CI ok

File: features/Context/DataGridContext.php [hunk 2]
```diff
@@ -404,6 +404,17 @@ public function iChangePageSize($size)
     }
 
     /**
+     * @param int $size
+     *
+     * @When /^page size should be (.+)$/
```
Comment (skeleton) [inline features/Context/DataGridContext.php]: "\d+" instead of ".+"

**Pull request merged**